public/components/freelancer-edit.js

```diff
@@ -0,0 +1,427 @@
+    return (
+      <li data-key={this.props.dataKey}>
+        {
+          date.toLocaleDateString('en-GB')
+          + " from " + begin.toLocaleTimeString('en-US',  timeSettings)
+          + " to " + end.toLocaleTimeString('en-US',  timeSettings)
+          + " in " + day.location
+        }
+        <input type="button" value="Delete" onClick={this.deleteDate} />
+      </li>
+    );
+  }
+}
+
+/**
+ * Helper and rendering functions
+ */
+
+function renderError(errorString) {
+  document.getElementById('emergency-single-date-error').innerHTML = errorString;
+}
+
+function renderPage() {
+  ReactDOM.render(<FreelancerEditView />, document.getElementById('react-freelancer-edit'));
+};
+
+/**
+ * Render list of single days
+ * @param  {Day[]} days array of days to display
+ */
+function renderSingleDates(days) {
+  // get single days (not set via weekly schedule)
+  savedSingleDates = days.filter((day) => { return !day.isRepeated; });
+
+  let dayList = savedSingleDates.map((day, index) => <FreelancerEmergencySingleDate day={day} key={index} dataKey={index} /> );
+  ReactDOM.render(
+    <ul>
+      <label>Single dates saved:</label>
+      {dayList}
+    </ul>,
+    document.getElementById('react-freelancer-emergency-single-list')
+  );
+}
+
+/**
+ * fill in checkboxes based on saved data
+ * @param  {Day[]} days to display (may also contain non-repeated ones, they are filtered)
+ */
+function renderRepeatedDates(days) {
+  savedRepeatedDates = days.filter((day) => { return day.isRepeated; });
+  let form = document.getElementById('emergency-form-repetition');
+  let displayUntil = document.getElementById('emergency-repetition-saved-until');
+  displayUntil.innerHTML = "";
+  savedRepeatedDates.forEach(function(day) {
+    let dayOfWeek = new Date(day.begin).getDay();
+    let end = document.getElementById('emergency-time-' + dayOfWeek + '-end');
+    let begin = document.getElementById('emergency-time-' + dayOfWeek + '-start');
+    let location = document.getElementById('emergency-location-' + dayOfWeek + '');
+    let check = end.parentNode.firstChild;
+    if(check.checked) {
+      let lastDay = savedRepeatedDates[savedRepeatedDates.length - 1];
+      displayUntil.innerHTML = "Repeats until " + new Date(lastDay.begin).toLocaleDateString('en-GB');
+      return; // already set this day of the week
+    } else {
+      check.checked = true;
+      end.disabled = false;
+      end.required = true;
+      begin.disabled = false;
+      begin.required = true;
+      location.disabled = false;
+      location.required = true;
+    }
+
+    end.value = toTimeString(new Date(day.end));
+    begin.value = toTimeString(new Date(day.begin));
+    location.value = day.location;
+  });
+}
+
+/**
+ * Sends AJAX request to save `savedSingleDates` + `savedRepeatedDates`
+ */
+function updateDates() {
+  let emergencyDates = savedSingleDates.concat(savedRepeatedDates);
+  console.log("savedRepeatedDates", savedRepeatedDates);
```

Comment:
Doesn't really matter (since this app never goes live anyways), but if you want, you can remove this log.